scipy/optimize/_minpack_py.py

```diff
@@ -547,7 +547,8 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     xdata : array_like or object
         The independent variable where the data is measured.
         Should usually be an M-length sequence or an (k,M)-shaped array for
-        functions with k predictors, but can actually be any object.
+        functions with k predictors, and each element should be float
+        convertible if it is an array like object.
```

Comment:
Agreed that "any object" is far too broad. I wonder if we should also remove the `or object` in the type info after `xdata` above as well? I don't think we normally use type info that broad in docstrings.